source/Xunit.Gherkin.Quick/Feature.cs

```diff
@@ -32,39 +30,52 @@ internal void Scenario(string scenarioName)
 
             var parsedScenario = gherkinDocument.Feature.Children.FirstOrDefault(scenario => scenario.Name == scenarioName);
             if (parsedScenario == null)
-                throw new Exception($"Cannot find scenario `{scenarioName}`.");
+                throw new GherkinException($"Cannot find scenario `{scenarioName}`.");
 
             var stepMethods = GetType().GetTypeInfo().GetMethods()
-                .Where(m => m.IsDefined(typeof(KeyWordAttributeBase)))
-                .Select(m => new { method = m, keywordAttribute = m.GetCustomAttribute<KeyWordAttributeBase>() });
+                .Where(m => m.IsDefined(typeof(StepDefinitionAttributeBase)))
+                .Select(m => new { method = m, keywordAttribute = m.GetCustomAttribute<StepDefinitionAttributeBase>() });
 
             foreach (var parsedStep in parsedScenario.Steps)
             {
                 var matchingStepMethod = stepMethods.FirstOrDefault(stepMethod => 
                     stepMethod.keywordAttribute.MatchesStep(parsedStep.Keyword, parsedStep.Text));
                 if (matchingStepMethod == null)
-                    throw new Exception($"Cannot find scenario step `{parsedStep.Keyword}{parsedStep.Text}` for scenario `{scenarioName}`.");
-
+                    throw new GherkinException($"Cannot find scenario step `{parsedStep.Keyword}{parsedStep.Text}` for scenario `{scenarioName}`.");
+                
                 var stepRegexMatch = matchingStepMethod.keywordAttribute.MatchRegex(parsedStep.Text);
                 if (!string.Equals(stepRegexMatch.Value, parsedStep.Text.Trim(), StringComparison.OrdinalIgnoreCase))
-                    throw new Exception($"Step method partially matched but not selected. Step `{parsedStep.Text.Trim()}`, Method pattern `{matchingStepMethod.keywordAttribute.Pattern}`.");
+                    throw new GherkinException($"Step method partially matched but not selected. Step `{parsedStep.Text.Trim()}`, Method pattern `{matchingStepMethod.keywordAttribute.Pattern}`.");
 
                 var methodParams = matchingStepMethod.method.GetParameters();
-                object[] methodParamValues = null;
-                if (methodParams.Length > 0)
+
+                var methodParamsCount = (methodParams.LastOrDefault()?.ParameterType == typeof(DataTable))
```

Comment:
Is this realistic to think that datatable should always be last? please refer to my comment for the previously existing PR. I think the position should not be necessary to be last.
Thoughts?